ethereumj-core/src/main/java/org/ethereum/sharding/crypto/BLS381.java

```diff
@@ -0,0 +1,53 @@
+package org.ethereum.sharding.crypto;
+
+import java.math.BigInteger;
+
+/**
+ * Common interface for BLS381-12 implementation
+ */
+public interface BLS381 {
+
+    BI generatePrivate();
+
+    BI restorePrivate(BigInteger value);
```

Comment:
I would abstract from any particular big number implementation. Wouldn't `byte[]` representation be sufficient enough?